src/main/java/org/commcare/resources/model/ResourceInstaller.java

```diff
@@ -102,7 +102,7 @@ boolean install(Resource r, ResourceLocation location,
      * @return True if the upgrade step was completed successfully.
      * @throws UnresolvedResourceException If the local resource definition could not be found
      */
-    boolean upgrade(Resource r) throws UnresolvedResourceException;
+    boolean upgrade(Resource r, T instance) throws UnresolvedResourceException;
```

Comment:
Can you rename all the places where you call this variable `instance` to `platform` instead? `instance` is so generic (could be an instance of anything)

Reply:
I knew you'd say that :|